### PR TITLE
Git_Basics.rst: Add 'Common Git Issues' section

### PR DESCRIPTION
**Pull request by SarthakNijhawan**
This section will help newcomers sort basic git issues.

Fixes: https://github.com/coala/coala/issues/3140

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by abhsag24** [comment]
@SarthakNijhawan can you run `coala` over your code ammend your commit and push again? these gitmate issues will vanish :)

**Comment by RohanVB** [comment]
@SarthakNijhawan Since your commit involves LineLengthBear being flagged, coala will only give a warning, and it won't have an automatic patch option. You'll have to adjust the line lengths manually first to be <80 chars. Please rebase first, make the changes, then push.

**Comment by SarthakNijhawan** [comment]
Ohk, Thanks for the feedback guys.... I'll change it ...right now... 😃

**Comment by sims1253** [comment]
I vote closes for this as this does not fix a bug but introduces a new "feature".
Change the Fixes in your commit message to Closes ([coala.io/commit](coala.io/commit) for more info)

File: docs/Developers/Git_Basics.rst [hunk 1]
```diff
@@ -283,6 +283,35 @@ another editor will be fired up with all the messages of commits which you want
 to squash. Clean up all the messages and add a new message to be
 displayed for the single commit.
 
+Common Git Issues
+-----------------
+Many a times , it might happen that after rebasing, you use ``git add -A`` which
```
Comment (sims1253) [inline docs/Developers/Git_Basics.rst]: space before comma

Comment (sims1253) [inline docs/Developers/Git_Basics.rst]: Usually it is "Many a time" without s though both seem to be viable.

File: docs/Developers/Git_Basics.rst [hunk 2]
```diff
@@ -283,6 +283,35 @@ another editor will be fired up with all the messages of commits which you want
...
+in turn commit those files into your branch and push it to the remote , you
+actually don't want to. Here ,is a short outline of, how can you remove (or
```
Comment (sims1253) [inline docs/Developers/Git_Basics.rst]: I would maybe rephrase the Sentence to something like:
"Sometimes it might happen that you use git add -A and add files you didn't want to."

File: docs/Developers/Git_Basics.rst [hunk 3]
```diff
@@ -283,6 +283,35 @@ another editor will be fired up with all the messages of commits which you want
...
+revert changes in) particular files from your commit even after pushing it to
+the remote .
+
+In your local repo , to revert the file to the state before the last commit run:
+::
+
+    $ git checkout HEAD^ /path/to/file
```
Comment (sims1253) [inline docs/Developers/Git_Basics.rst]: what does the ^ do ? never used it :D

File: docs/Developers/Git_Basics.rst [hunk 4]
```diff
@@ -283,6 +283,35 @@ another editor will be fired up with all the messages of commits which you want
...
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
+Besides, the ``git checkout`` command exhibits more flexibility in reverting the
+changes in a file, done even before the last commit , by replacing the ``HEAD^``
+in the above command by the revision number of the particular HEAD commit, you
+want to refer to.
+
```
Comment (sims1253) [inline docs/Developers/Git_Basics.rst]: Maybe add an example. This is for newcomers so they might not know what exactly to do here.

File: docs/Developers/Git_Basics.rst (hunk 2 above)
Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: Ok, I'll change it

File: docs/Developers/Git_Basics.rst (hunk 3 above)
Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: I saw the same in the stackOverflow answer , worked out for me , so added it .. 😛

File: docs/Developers/Git_Basics.rst (hunk 4 above)
Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: Ok , I'll add an example too..

File: docs/Developers/Git_Basics.rst (hunk 3 above)
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: @sims1253 it expands to one commit before the given ref

File: docs/Developers/Git_Basics.rst [hunk 5]
```diff
@@ -283,6 +283,50 @@ another editor will be fired up with all the messages of commits which you want
...
+Now , after reverting the file(s) update your last commit , by running :
+::
+
+    $ git commit --amend
+
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
```
Comment (Udayan12167) [inline docs/Developers/Git_Basics.rst]: Add a not here that my fork is name of the remote of your fork.

**Comment by Udayan12167** [comment]
Are you going to add more common git issues to this?

**Comment by SarthakNijhawan** [comment]
@Udayan12167 As far as the issue is concerned I have added the essentials and the necessary , but if more of the common issues are reported , I'll add it or can be opened as a separate issue too .
i'm ready add , if reported .

**Comment by Mixih** [comment]
@Udayan12167 There are a few other open issues that need this section, hence why I suggested it

File: docs/Developers/Git_Basics.rst [hunk 6]
```diff
@@ -283,6 +283,54 @@ another editor will be fired up with all the messages of commits which you want
 to squash. Clean up all the messages and add a new message to be
 displayed for the single commit.
 
+Common Git Issues
+-----------------
+Sometimes, it might happen that after rebasing, you use ``git add -A`` and
+add files you did'nt want to and push it to the remote. Here ,is a short outline
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: did'nt -> didn't

File: docs/Developers/Git_Basics.rst [hunk 7]
```diff
@@ -283,6 +283,54 @@ another editor will be fired up with all the messages of commits which you want
 to squash. Clean up all the messages and add a new message to be
 displayed for the single commit.
 
+Common Git Issues
+-----------------
+Sometimes, it might happen that after rebasing, you use ``git add -A`` and
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: I would change this to -> "Sometimes, you use git ``add-A`` and add files you didn't want to your push (often after rebasing)"

File: docs/Developers/Git_Basics.rst [hunk 8]
```diff
@@ -283,6 +283,54 @@ another editor will be fired up with all the messages of commits which you want
...
+add files you did'nt want to and push it to the remote. Here ,is a short outline
+of, how can you remove (or revert changes in) particular files from your commit
+even after pushing it to the remote .
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: remove "it" and "the"

File: docs/Developers/Git_Basics.rst [hunk 9]
```diff
@@ -283,6 +283,54 @@ another editor will be fired up with all the messages of commits which you want
...
+
+In your local repo , to revert the file to the state before the last commit run:
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: last -> previous
Make a very obvious note that this only rolls back one commit

File: docs/Developers/Git_Basics.rst [hunk 10]
```diff
@@ -283,6 +283,50 @@ another editor will be fired up with all the messages of commits which you want
...
+
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
+Besides, the ``git checkout`` command offers you more flexibility in reverting
+the changes in a file, done even from earlier than the last commit. By replacing
+the ``HEAD^`` by the revision number of the particular HEAD commit, you can
+refer to the required revision of the file.
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: It should, but that can become messy

File: docs/Developers/Git_Basics.rst [hunk 11]
```diff
@@ -283,6 +283,54 @@ another editor will be fired up with all the messages of commits which you want
...
+
+    myfork is your forked repository , where you push your commits.
+
+Besides, the ``git checkout`` command offers you more flexibility in reverting
+the changes in a file, done even from earlier than the last commit. By replacing
+the ``HEAD^`` by the revision number of the particular HEAD commit, you can
+refer to the required revision of the file.
+
+Might sound a little intimidating, but don't worry, an example has been
+illustrated for your reference.
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: illustrated -> provided, your reference -> you

File: docs/Developers/Git_Basics.rst [hunk 12]
```diff
@@ -283,6 +283,54 @@ another editor will be fired up with all the messages of commits which you want
...
+First you can check the commit's revision number, where the file was revised by
+running the following command:
+::
+
+    $ git log /path/to/file
+
+The revision number might look like ``3cdc61015724f9965575ba954c8cd4232c8b42e4``
+Now, to revert the file to that stage , run the command:
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: stage -> commit(or revision) (or just use changes rolled back)

File: docs/Developers/Git_Basics.rst [hunk 13]
```diff
@@ -283,6 +283,54 @@ another editor will be fired up with all the messages of commits which you want
...
+::
+
+    $ git checkout 3cdc61015724f9965575ba954c8cd4232c8b42e4 /path/to/file.txt
+
+Now, after the file gets reverted back to the required stage ,commit the changes
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: stage -> commit(or revision) (or just use changes rolled back)

File: docs/Developers/Git_Basics.rst (hunk 6 above)
Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: Oh yea sorry , my bad... 😛

File: docs/Developers/Git_Basics.rst [hunk 14]
```diff
@@ -283,6 +283,58 @@ another editor will be fired up with all the messages of commits which you want
...
+Might sound a little intimidating, but don't worry, an example has been
+provided for you.
+First you can check the commit's revision number, where the file was revised by
+running the following command:
+::
+
+    $ git log /path/to/file
+
+The revision number might look like ``3cdc61015724f9965575ba954c8cd4232c8b42e4``
+Now, to revert the file to that commit(or revision) (or just use changes rolled
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: hmm sorry, I wasn't clear, I meant you should have chosen one of the changes I suggested here, not all three :)

Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: The same goes for the second location of those

Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: Hahaha .... I'll change it though .... Btw the newcomers can choose any one of them too.. :stuck_out_tongue:

File: docs/Developers/Git_Basics.rst [hunk 15]
```diff
@@ -283,6 +283,57 @@ another editor will be fired up with all the messages of commits which you want
...
+    $ git commit --amend
+
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
+.. note::
+
+    The upstated procedure helps roll back changes by one commit only.
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: upstated procedure -> procedure outlined above
(gah, sorry I missed, it, it was hidden by a massive comment chain.

File: docs/Developers/Git_Basics.rst [hunk 16]
```diff
@@ -283,6 +283,57 @@ another editor will be fired up with all the messages of commits which you want
...
+
+    $ git log /path/to/file
+
+The revision number might look like ``3cdc61015724f9965575ba954c8cd4232c8b42e4``
+Now, to revert the file to that revision, run the command:
+::
+
+    $ git checkout 3cdc61015724f9965575ba954c8cd4232c8b42e4 /path/to/file.txt
+
+Now, after the file gets reverted back to the required revision,commit the
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: "revision,commit" needs a space

**Comment by Mixih** [comment]
Oh and you might need to rebase

**Comment by SarthakNijhawan** [comment]
Ok , I'll do it ...

File: docs/Developers/Git_Basics.rst [hunk 17]
```diff
@@ -283,6 +283,57 @@ another editor will be fired up with all the messages of commits which you want
...
+::
+
+    $ git push -f myfork
+
+.. note::
+
+    The procedure outlined above helps roll back changes by one commit only.
+    Besides, myfork is your forked repository, where you push your commits.
+
+Besides, the ``git checkout`` command offers you more flexibility in reverting
```
Comment (Nosferatul) [inline docs/Developers/Git_Basics.rst]: Could we specify that if you use `git checkout <file>` revert the changes made in the file?

Comment (Nosferatul) [inline docs/Developers/Git_Basics.rst]: Everything looks good to me

File: docs/Developers/Git_Basics.rst [hunk 18]
```diff
@@ -283,6 +283,57 @@ another editor will be fired up with all the messages of commits which you want
...
+
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
+.. note::
+
+    The procedure outlined above helps roll back changes by one commit only.
+    Besides, myfork is your forked repository, where you push your commits.
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: Hmmm, please remove "Besides" as it is not an explanation or justification of why it's a one commit removal.

**Comment by Mixih** [comment]
Two changes, and a rebase, then we can push!

File: docs/Developers/Git_Basics.rst (hunk 18 above)
Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: Done!

File: docs/Developers/Git_Basics.rst [hunk 19]
```diff
@@ -283,6 +283,58 @@ another editor will be fired up with all the messages of commits which you want
...
+
+    $ git push -f myfork
+
+.. note::
+
+    The procedure outlined above helps roll back changes by one commit only.
+    The above mentioned, 'myfork' is your forked repository, where you push your
+    commits.
+
+Besides, the ``git checkout`` command offers you more flexibility in reverting
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: Delete "besides"... Also ``git checkout`` -> ``git checkout <revision sha> /path/to/file`` ?
(per nosferatul's suggestion)

Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: Oh , sorry I forgot.

File: docs/Developers/Git_Basics.rst [hunk 20]
```diff
@@ -283,6 +283,58 @@ another editor will be fired up with all the messages of commits which you want
...
+
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
+.. note::
+
+    The procedure outlined above helps roll back changes by one commit only.
+    The above mentioned, 'myfork' is your forked repository, where you push your
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: `The above mentioned,` No comma here please

Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: 👍

**Comment by Mixih** [comment]
it's ready. Just a rebase please

**Comment by Mixih** [comment]
oh and while you do that, change sort in the commit body to "sort out" thanks!

**Comment by SarthakNijhawan** [comment]
@Mixih Done!

**Comment by yukiisbored** [comment]
ack 7699622

**Comment by Mixih** [comment]
sorry, one more rebase. When you do that, immediately go into the gitter and ping me.

File: docs/Developers/Git_Basics.rst [hunk 21]
```diff
@@ -283,6 +283,58 @@ another editor will be fired up with all the messages of commits which you want
...
+In your local repo, to revert the file to the state before the previous commit
+run the following:
+::
+
+    $ git checkout HEAD^ /path/to/file
+
+Now , after reverting the file(s) update your last commit , by running :
+::
+
+    $ git commit --amend
```
Comment (Mixih) [inline docs/Developers/Git_Basics.rst]: I think we need the -a so it automatically stages all of the modified files.

Comment (SarthakNijhawan) [inline docs/Developers/Git_Basics.rst]: On it!

**Comment by Mixih** [comment]
ack 38206d0

**Comment by Mixih** [comment]
@rultor merge
